orc/src/main/java/org/apache/iceberg/orc/OrcIterable.java

```diff
@@ -59,10 +59,17 @@
   @SuppressWarnings("unchecked")
   @Override
   public Iterator<T> iterator() {
+    final Reader orcFileReader = newFileReader(file, config);
+    final ColumnMap colMapping = orcFileReader.hasMetadataValue(ORC.COLUMN_NUMBERS_ATTRIBUTE) ?
+        ColumnMap.deserialize(orcFileReader.getSchema(),
+            orcFileReader.getMetadataValue(ORC.COLUMN_NUMBERS_ATTRIBUTE)) : new ColumnMap();
```

Comment:
This is a good place for a "TODO" comment for adding a default mapping. In Parquet, for example, when there are no IDs, we assign IDs using position to read older data. This would also need to check whether each column has a null value and set the required/optional flag for the column correctly.

Reply:
Yeah, this part creates a default column mapping using position (relative to the flattened ORC schema) as well to assign IDs. I've added a `TODO` item to add the checks for nulls to determine whether columns are `required` or `optional`.